WalletWasabi/WabiSabi/Backend/Rounds/Arena.Partial.cs

```diff
@@ -358,11 +357,14 @@ public async Task<Coin> OutpointToCoinAsync(InputRegistrationRequest request, Ca
 		return new Coin(input, txOutResponse.TxOut);
 	}
 
-	public async Task<RoundState[]> GetStatusAsync(CancellationToken cancellationToken)
+	public async Task<RoundState[]> GetStatusAsync(RoundStateRequest request, CancellationToken cancellationToken)
 	{
 		using (await AsyncLock.LockAsync(cancellationToken).ConfigureAwait(false))
 		{
-			return Rounds.Select(x => RoundState.FromRound(x)).ToArray();
+			return Rounds.Select(x => {
+				var checkPoint = request.RoundCheckpoints.FirstOrDefault(y => y.RoundId == x.Id);
```

Comment:
This code is fine and works as it should. However, later we can strip down more information by sending only the requested rounds. Now if a client misses one round it will get all the information (only once at the beginning), no matter what.